### PR TITLE
tests: fix unstable tests

### PR DESCRIPTION
**Pull request by lance6716**
### What problem does this PR solve? 
part of #1790 

### What is changed and how it works?
- don't check result of resume. Because if worker runs quickly, resume-task will see the expected error of next query-status and not return "result: true"
- wait DM-worker clean dump files and then using `ls` to check files
- also check master peer port before restart master, to avoid port in use

### Check List 

Tests 

 - No code

Related changes

 - Need to cherry-pick to the release branch

### CODE REVIEW

File: tests/sharding/run.sh [hunk 1]
```diff
@@ -80,17 +80,15 @@ function run() {
...
+		"resume-task test"
 
 	run_dm_ctl_with_retry $WORK_DIR "127.0.0.1:$MASTER_PORT" \
 		"query-status test" \
 		"failpoint error for FlushCheckpointStage before track DDL" 1
 
 	# resume-task to next stage
 	run_dm_ctl $WORK_DIR "127.0.0.1:$MASTER_PORT" \
-		"resume-task test" \
-		"\"result\": true" 3
```
Comment (Ehco1996) [inline tests/sharding/run.sh:93]: why don't you need to test the number of `result: true`
Is it because the following is tested with `dmctl query-status`?

Comment (lance6716) [reply tests/sharding/run.sh:93]: see PR description

**Review by Ehco1996 — APPROVED** [review]

**Review by GMHDBJD — APPROVED** [review]
LGTM

**Pull request merged**